Orange/data/table.py

```diff
@@ -204,11 +229,22 @@ def __new__(cls, *args, **kwargs):
         return cls.from_numpy(domain, *args, **kwargs)
 
     def __init__(self, *args, **kwargs):
-        # So subclasses can expect to call super without breakage; noop
-        pass
+        # TODO: use this instead of __new__
+        # all weights initialized to 1 (see the weight functions for details)
+        self[Table._WEIGHTS_COLUMN] = 1
+
+        self.name = kwargs.get("name") or self.name
+
+        # used for differentiating columns into x/y/meta, as a pandas property
+        self._columns_X = set()
+        self._columns_Y = set()
+        self._columns_meta = set()
 
     @classmethod
     def from_domain(cls, domain, n_rows=0, weights=False):
+        # TODO: change
+        # TODO: do we really need to init with zeroes? also, how does this work with string variables etc
+        # TODO: init with nas I guess
```

Comment:
Nobody uses this except some Table's methods and tests. You replace those; you find everything still works.